addon/services/intl.js

```diff
@@ -93,14 +88,17 @@ export default Service.extend(Evented, {
 
   /** @public **/
   init() {
-    this._super();
+    this._super(...arguments);
 
     if (typeof Intl === 'undefined') {
       warn(`[ember-intl] Intl API was not found.\nSee: ${links.polyfill}`, false, {
         id: 'ember-intl-undefined-intljs'
       });
     }
 
+    const initialLocale = get(this, 'locale') || ['en-us'];
+
+    this.setLocale(initialLocale);
```

Comment:
I plan to do more work in the future around this method.

* renaming
* moving locale setter into setLocale (again)
* deprecate & remove locale setter
* consider asserting provided array (for consistency and can make better downstream assumptions)